plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,811 @@
+  object_name:
+    description:
+      - The name of the object.
+      - If not specified, a list of all objects in the specified bucket will be returned.
+    required: false
+    type: str
+  endpoint_url:
+    description:
+      - S3 URL endpoint for usage with Ceph, Eucalyptus and fakes3 etc. Otherwise assumes AWS.
+    type: str
+  dualstack:
+    description:
+      - Enables Amazon S3 Dual-Stack Endpoints, allowing S3 communications using both IPv4 and IPv6.
+    type: bool
+    default: false
+  ceph:
+    description:
+      - Enable API compatibility with Ceph RGW.
+      - It takes into account the S3 API subset working with Ceph in order to provide the same module
+        behaviour where possible.
+      - Requires I(endpoint_url) if I(ceph=true).
+    aliases: ['rgw']
+    default: false
+    type: bool
+  object_details:
+    description:
+      - Retrieve requested S3 object detailed information.
+    required: false
+    type: dict
+    suboptions:
+      object_acl:
+        description:
+          - Retreive S3 object ACL.
+        required: false
+        type: bool
+        default: false
+      object_legal_hold:
+        description:
+          - Retreive S3 object legal_hold.
+        required: false
+        type: bool
+        default: false
+      object_lock_configuration:
+        description:
+          - Retreive S3 object lock_configuration.
+        required: false
+        type: bool
+        default: false
+      object_retention:
+        description:
+          - Retreive S3 object retention.
+        required: false
+        type: bool
+        default: false
+      object_tagging:
+        description:
+          - Retreive S3 object Tags.
+        required: false
+        type: bool
+        default: false
+      object_attributes:
+        description:
+          - Retreive S3 object attributes.
+          - Requires minimum botocore version 1.24.7.
+        required: false
+        type: bool
+        default: false
+      attributes_list:
+        description:
+          - The fields/details that should be returned.
+          - Required when I(object_attributes) is C(true) in I(object_details).
+        type: list
+        elements: str
+        choices: ['ETag', 'Checksum', 'ObjectParts', 'StorageClass', 'ObjectSize']
+notes:
+  - Support for I(tags) and I(purge_tags) was added in release 2.0.0.
+  - In release 5.0.0 the I(s3_url) parameter was merged into the I(endpoint_url) parameter,
+    I(s3_url) remains as an alias for I(endpoint_url).
+  - For Walrus I(endpoint_url) should be set to the FQDN of the endpoint with neither scheme nor path.
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+
+'''
+
+EXAMPLES = r'''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Retrieve a list of objects in S3 bucket
+  amazon.aws.s3_object_info:
+    bucket_name: MyTestBucket
+
+- name: Retrieve a list of objects in Ceph RGW S3
```

Comment:
The same. I don't think it's a good idea to demonstrate how to use the module with a use-case we don't support.